### ISSUE TITLE
add explicit `name` keyword for Action?

### ISSUE THREAD

**Issue by hainm** [Amber-MD/cpptraj#186]
Not heard from this issue yet (#100), so I make separate issue.

For most of  Actions, we don't need to write down the word `name` but for Analysis, we can.

What's your opinion about adding name for Dataset in Action too?

```python
rms first name my_rmsd
```
which is equal to 

```python
rms my_rmsd first
```

**note**: Some Action using explicit name keyword: Action_Principal.cpp, Action_NativeContacts.cpp, Action_Matrix

```bash
 mprintf("\t[name <name>] [ byatom | byres [mass] | bymask [mass] ]\n"
```

This is easy task, so if you guys are ok, I can add `name` (still make backward compat though).

**Comment by drroe** [comment]
> What's your opinion about adding name for Dataset in Action too?

In my opinion this falls under the umbrella of the old adage, "If it ain't broke, don't fix it". The overall intent of what you want to do, which I believe is to make the syntax more uniform between commands, is nice. However, I think you're underestimating the amount of work involved. *Every single Action* like `rms` will have to have some variation of the following code added, e.g.
```C++
std::string setname = actionArgs.GetStringKey("name");
...
MetaData md;
if (setname.empty())
  md = MetaData( actionArgs.GetStringNext(), MetaData::M_RMS );
else
  md = MetaData( setname, MetaData::M_RMS );
```
The above code won't work exactly for lots of Actions (`dssp` for example would need something completely different) so each Action will have to have some slight customization of the above code, not to mention testing to make sure backwards compatibility is not broken.

The final problem is that `name` already sometimes means something different, e.g. in `modes` analysis `name` specifies the eigenmodes data set you want to *use* for calculation.

So in summary, I think this is lots of work for minimal payoff. However, Maybe the developers guide can be updated to suggest that for future Actions/Analyses `name` be reserved for naming created DataSets.

**Comment by hainm** [comment]
> However, I think you're underestimating the amount of work involved.

> So in summary, I think this is lots of work for minimal payoff.

thanks. So I give up.

>  However, Maybe the developers guide can be updated to suggest that for future Actions/Analyses name be reserved for naming created DataSets.

:+1:
